### ISSUE TITLE
build failure since #360

### ISSUE THREAD

**Issue by klaernie** [editorconfig-checker/editorconfig-checker#397]
I tried to build this package with this PKGBUILD and this pr (since this pr to be precise)
```shell
pkgname=editorconfig-checker-git
_pkgname=editorconfig-checker
_commit=cf4ae1ccede331b2aa1b115f1de5257737de7eef
pkgver=next
pkgrel=1
pkgdesc='A tool to verify that your files are in harmony with your .editorconfig'
arch=(x86_64)
url="https://github.com/editorconfig-checker/$_pkgname"
license=(MIT)
depends=(glibc)
makedepends=(go)
provides=(ec)
conflicts=("$_pkgname")
source=(
        "git+${url}.git#commit=$_commit"
)
sha256sums=('SKIP')

build() {
        cd "$srcdir/$_pkgname"
        export CGO_CPPFLAGS="$CPPFLAGS"
        export CGO_CFLAGS="$CFLAGS"
        export CGO_CXXFLAGS="$CXXFLAGS"
        export CGO_LDFLAGS="$LDFLAGS"
        export GOFLAGS="-buildmode=pie -trimpath -ldflags=-linkmode=external -mod=readonly -modcacherw" # won't pass check
        # export GOFLAGS="-buildmode=pie -ldflags=-linkmode=external -mod=readonly -modcacherw" # will pass check
        go build -o "$_pkgname" ./cmd/...
}

check() {
        cd "$srcdir/$_pkgname"
        go test ./...
}

package() {
        cd "$srcdir/$_pkgname"
        install -Dm755 -t "$pkgdir/usr/bin/" "$_pkgname"
        ln -sf "$_pkgname" "$pkgdir/usr/bin/ec"
        install -Dm644 -t "$pkgdir/usr/share/licenses/$_pkgname" LICENSE
}

```
I got this error
```
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating editorconfig-checker git repo...
==> Validating source files with sha256sums...
    editorconfig-checker ... Skipped
==> Extracting sources...
  -> Creating working copy of editorconfig-checker git repo...
Cloning into 'editorconfig-checker'...
done.
Switched to a new branch 'makepkg'
==> Starting build()...
==> Starting check()...
?       github.com/editorconfig-checker/editorconfig-checker/v3/cmd/editorconfig-checker        [no test files]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/config      0.004s
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/encoding    0.034s
--- FAIL: TestFormatErrors (0.00s)
    error_test.go:320: mkdir github.com: permission denied
    error_test.go:323: mkdir github.com: permission denied
    error_test.go:326: mkdir github.com: permission denied
FAIL

Snapshot Summary

✕ 3 snapshots failed

FAIL    github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error       0.003s
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files       0.009s
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/logger      (cached)
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/utils       0.002s
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation  0.017s
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation/validators       (cached)
FAIL
==> ERROR: A failure occurred in check().
    Aborting...
```

If I remove `-trimpath` in `GOFLAGS`, it can build fine. Is it ok if I (or next archlinux official package) just remove this flag?

_Originally posted by @xsy420 in https://github.com/editorconfig-checker/editorconfig-checker/issues/360#issuecomment-2460755678_

**Comment by klaernie** [comment]
That flag is [necessary](https://wiki.archlinux.org/title/Go_package_guidelines#Flag_meaning) for the reproducibility. I just checked and the package builds fine in a [clean chroot](https://wiki.archlinux.org/title/DeveloperWiki:Building_in_a_clean_chroot) - you can remove the flag if it causes problems in your local machine :)

_Originally posted by @orhun in https://github.com/editorconfig-checker/editorconfig-checker/issues/360#issuecomment-2471526452_

**Comment by klaernie** [comment]
@orhun Really? I also tried in gitlab cicd.
See https://gitlab.archlinux.org/xsy420/editorconfig-checker-git/-/pipelines/112082

_Originally posted by @xsy420 in https://github.com/editorconfig-checker/editorconfig-checker/issues/360#issuecomment-2480833798_

**Comment by klaernie** [comment]
> I also tried in gitlab cicd. See https://gitlab.archlinux.org/xsy420/editorconfig-checker-git/-/pipelines/112082

I noticed you set the build options using `$GOFLAGS`, including `-trimpath`. What I notice from the Gitlab CI run and now also see in the PKGBUILD: `$GOFLAGS` applies both to `go build` and `go test`. And what fails are just the test cases using snapshot testing with [go-snaps](https://github.com/gkampitakis/go-snaps)

I can reproduce this locally:
```
kandre@mainframe(pts/5)[main] ~/git-repos/editorconfig-checker % go test -trimpath -run ^TestFormatErrors$ github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error  
--- FAIL: TestFormatErrors (0.00s)
    --- FAIL: TestFormatErrors/default (0.00s)
        error_test.go:320: mkdir github.com: permission denied
    --- FAIL: TestFormatErrors/codeclimate (0.00s)
        error_test.go:320: mkdir github.com: permission denied
    --- FAIL: TestFormatErrors/gcc (0.00s)
        error_test.go:320: mkdir github.com: permission denied
    --- FAIL: TestFormatErrors/github-actions (0.00s)
        error_test.go:320: mkdir github.com: permission denied
FAIL

Snapshot Summary

✕ 4 snapshots failed

FAIL    github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error       0.004s
FAIL
kandre@mainframe(pts/5)[main] ~/git-repos/editorconfig-checker % go test -run ^TestFormatErrors$ github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error  
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error       0.005s
kandre@mainframe(pts/5)[main] ~/git-repos/editorconfig-checker % git rev-parse HEAD
d1b82344dca8f48330b851a47570fb9b760e8387
kandre@mainframe(pts/5)[main] ~/git-repos/editorconfig-checker %
```

This leads down to the next piece that needs investigation: why does `go test -trimpath` fail. Interesting to compare the case by pretending to run in a CI environment, where the issue becomes a bit more apparent:

```
kandre@mainframe(pts/5)[main] ~/git-repos/editorconfig-checker % CI=true go test -trimpath -run ^TestFormatErrors$ ./...   
?       github.com/editorconfig-checker/editorconfig-checker/v3/cmd/editorconfig-checker        [no test files]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/config      (cached) [no tests to run]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/encoding    (cached) [no tests to run]
--- FAIL: TestFormatErrors (0.00s)
    --- FAIL: TestFormatErrors/default (0.00s)
        error_test.go:320: snapshot not found
    --- FAIL: TestFormatErrors/codeclimate (0.00s)
        error_test.go:320: snapshot not found
    --- FAIL: TestFormatErrors/gcc (0.00s)
        error_test.go:320: snapshot not found
    --- FAIL: TestFormatErrors/github-actions (0.00s)
        error_test.go:320: snapshot not found
FAIL

Snapshot Summary

✕ 4 snapshots failed

FAIL    github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error       0.004s
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files       (cached) [no tests to run]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/logger      (cached) [no tests to run]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/outputformat        (cached) [no tests to run]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/utils       (cached) [no tests to run]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation  (cached) [no tests to run]
ok      github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation/validators       (cached) [no tests to run]
FAIL
kandre@mainframe(pts/5)[main] ~/git-repos/editorconfig-checker % 
```

It fails to find the snapshots. To debug I tried this under  `strace -e file -f -- go test -trimpath …`, and found it tried to find the `__snapshots__` directory in `/`.

If I now comment out this segment https://github.com/editorconfig-checker/editorconfig-checker/blob/18a2b1f9da0fd5f85141063dd16e8cf2aa1dc10a/pkg/error/error_test.go#L245-L247 and do NOT pretend to be a CI environment the `go test -trimpath` works again, but it autocreates the __snapshots__ directory.

So my conclusion is, that (at least right now) running tests with go-snaps under `-trimpath` is not supported, and we should build a minimal reproduction and report this upstream. But it might be, that this is unfixable, and we simply should not run the test suite with `-trimpath`

**Comment by theoludwig** [comment]
Thank you @klaernie, for looking into it.
I've had little free time last few weeks, to check what is going on with editorconfig-checker. 

Is this issue considered blocking for releasing v3.1.0?

@mstruebing Are we releasing v3.1.0 before 2024 ends, or we keep that release for early 2025, to start the year nicely? :smile:

**Comment by klaernie** [comment]
This issue only breaks the test suite when using `-trimpath`. My personal option would be a "don't do that", and none of our build infrastructure does that, and there is nothing to gain from enabling `-trimpath` while running the testsuite.

**Comment by mstruebing** [comment]
> @mstruebing Are we releasing v3.1.0 before 2024 ends, or we keep that release for early 2025, to start the year nicely? 😄

I don't really mind. I'm up for it if you need me to do some manual steps but I'm not confident in merging and running the automatic release process myself as I didn't build it and don't 100% know what to expect/how to fix it.

I can help or be in a call with someone, but I would like to give that power to you people actually :)

Or do I understand you wrong and we need a release of the current `main`-branch?

**Comment by theoludwig** [comment]
You understood well, I meant to merge #385 and running the automatic release process.
Yeah, I've never used GoReleaser en release-please myself too.

> I can help or be in a call with someone

Would be great to have a call with @klaernie, I don't know when/with what software? Discord? I don't think it needs to be lengthy anyway.

I don't mind personally, if somehow, we can't/don't want yet to use the automatic release process, manual release of the current `main` will do too.
I just thought there were enough improvements that were merged recently, that would be good that users could start using, so might be time to do a release.

**Comment by theoludwig** [comment]
@ccoVeille Do you want to join us for a call?

**Comment by klaernie** [comment]
@theoludwig sorry I didn't reply lately. I would suggest just merging #385 and failing forward in the unlikely case that something goes wrong. In the meantime I had the chance to test a similar goreleaser deployment at work, which worked nicely. But we should discuss this in #385 preferable. Call would probably best using a public Jitsi meet, although Discord would do as well in a pinch.

**Comment by anjandev** [comment]
Hi, I am packaging this program and we can reproduce this issue on alpine:

https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/78843#note_475506
